### PR TITLE
remove .sort / .reverse properties and adjust tests

### PR DESCRIPTION
**Pull request by UplinkCoder**
as per title
This is a revamp of #6554

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
Doesn't `idgen` need an update as well ?

**Comment by UplinkCoder** [comment]
@mathias-lang-sociomantic I originally intended for the idgen change to be in a separate PR
but it might as well happen now.

**Comment by UplinkCoder** [comment]
@andralex ready for merge.

**Review by andralex — APPROVED** [review]
thx!

**Comment by andralex** [comment]
cc @MartinNowak for deployment as he sees fit

**Comment by UplinkCoder** [comment]
There are still a few tests to fix up..
Should be ready soon though.

**Comment by UplinkCoder** [comment]
@MartinNowak this passes now.

File: test/runnable/foreach4.d [hunk 1]
```diff
@@ -840,7 +840,6 @@ void test26()
     assert(i == 2);
     assert(a.array[0] == 73);
     assert(a.array[1] == 82);
-
```
Comment (yebblies) [inline test/runnable/foreach4.d]: ??

File: test/runnable/test12.d [hunk 2]
```diff
@@ -941,27 +941,29 @@ void test44()
...
     {
+        import std.algorithm;
         return std.algorithm.cmp(this.str, s.str);
     }
 }
 
 void test45()
 {
+    import std.algorithm;
+    import std.range;
```
Comment (yebblies) [inline test/runnable/test12.d]: ??

File: test/runnable/test4.d [hunk 3]
```diff
@@ -1053,15 +997,6 @@ void test41()
 
 /* ================================ */
 
-void test42()
-{
-        real[10] array;
-        real[] copy = array.dup;
-        copy.sort;
```
Comment (yebblies) [inline test/runnable/test4.d]: Just delete this line?

File: test/runnable/wc2.d [hunk 4]
```diff
@@ -66,7 +66,7 @@ int main (string[] args)
     }
 
     printf("--------------------------------------\n");
-
+    import std.algorithm.sorting : sort;
```
Comment (yebblies) [inline test/runnable/wc2.d]: Nooooo

**Comment by UplinkCoder** [comment]
@yebblies addressed your comments

**Review by MartinNowak — CHANGES_REQUESTED** [review]
Short changelog notice, otherwise good to go from my side.
https://github.com/dlang/dmd/tree/master/changelog

**Comment by andralex** [comment]
thx you guys - @UplinkCoder changelog and we should be good to go

**Comment by UplinkCoder** [comment]
@andralex Martin amended the changelog entry already
You can dis-spell his request for changes.

**Review by MartinNowak — APPROVED** [review]

**Comment by wilzbach** [comment]
PR for dlang.org: https://github.com/dlang/dlang.org/pull/1672

**Pull request merged**